### PR TITLE
Risk by site

### PR DESCRIPTION
**Pull request by micheles**
Closes https://github.com/gem/oq-risklib/issues/729.

The demos are running here: https://ci.openquake.org/job/zdevel_oq-risklib/1408

### CODE REVIEW

File: openquake/calculators/views.py [hunk 1]
```diff
@@ -425,23 +425,35 @@ def view_assetcol(token, dstore):
...
+    n_sites = len(dstore['sitecol'].complete)
+    rlzs_assoc = dstore['rlzs_assoc']
+    num_ruptures = dstore.get_attr('tags', 'num_ruptures')
+    col = num_ruptures.argmax()
+    n_ruptures = num_ruptures[col]
+    trt_id = rlzs_assoc.csm_info.get_trt_id(col)
+    gsims = rlzs_assoc.gsims_by_trt_id[trt_id]
+    n_imts = len(oq.imtls)
+    n_rlzs = max(len(rlzs_assoc[trt_id, gsim]) for gsim in gsims)
+    size = n_sites * n_rlzs * n_ruptures * n_imts * 4
```
Comment (ptormene) [inline openquake/calculators/views.py]: Please add a comment about `4`

File: openquake/risklib/riskmodels.py [hunk 2]
```diff
@@ -714,21 +686,21 @@ def __init__(self, taxonomy, fragility_functions):
         self.taxonomy = taxonomy
         self.risk_functions = fragility_functions
 
-    def __call__(self, loss_type, assets, gmfs, _epsilons=None, _rupids=None):
+    def __call__(self, loss_type, assets, gmvs, _epsilons=None, _rupids=None):
         """
         :param loss_type: the loss type
         :param assets: a list of N assets of the same taxonomy
-        :param gmfs: an array of N x E elements
+        :param gmfs: an array of E elements
```
Comment (ptormene) [inline openquake/risklib/riskmodels.py]: gmfs -> gmvs

Comment (ptormene) [inline openquake/risklib/riskmodels.py]: you could document also _epsilons and _rupids

File: openquake/risklib/riskmodels.py [hunk 3]
```diff
@@ -617,6 +589,7 @@ def __init__(self, taxonomy,
         self.lrem_steps_per_interval = lrem_steps_per_interval
 
     def __call__(self, loss_type, assets, hazard, _eps=None, _rupids=None):
```
Comment (ptormene) [inline openquake/risklib/riskmodels.py]: missing docstring

File: openquake/risklib/riskmodels.py [hunk 4]
```diff
@@ -404,17 +403,17 @@ def __call__(self, loss_type, assets, hazard_curves, _epsilons=None,
             assets is an iterator over N
             :class:`openquake.risklib.scientific.Asset` instances
         :param hazard_curves:
```
Comment (ptormene) [inline openquake/risklib/riskmodels.py]: hazard_curves -> hazard_curve

**Comment by ptormene** [comment]
Besides minor comments, LGTM

File: openquake/calculators/views.py (hunk 1 above)
Comment (micheles) [inline openquake/calculators/views.py]: Done

File: openquake/risklib/riskmodels.py (hunk 2 above)
Comment (micheles) [inline openquake/risklib/riskmodels.py]: Done

File: openquake/risklib/riskmodels.py (hunk 3 above)
Comment (micheles) [inline openquake/risklib/riskmodels.py]: Added

File: openquake/risklib/riskmodels.py (hunk 4 above)
Comment (micheles) [inline openquake/risklib/riskmodels.py]: Fixed

**Pull request merged**